[Soo Jia Xi] iP

joosxi opened the pull request:
### Duke

> "Software is a great combination between artistry and engineering." - Bill Gates ([Poster with quote](https://www.brainyquote.com/quotes/bill_gates_626107#:~:text=Bill%20Gates%20Quotes&text=Software%20is%20a%20great%20combination%20between%20artistry%20and%20engineering.))

Duke is a **chatbot** that can help _record down your tasks!_ Our current and future features are:
- [ ] add either a "todo" task, an "event" task or a "deadline" task!
- [ ] "event" and "deadline" tasks allow you to add in dates and timings!
- [ ] mark the task as "done" when you have completed it, or "undone" if you have wrongly marked it!
- [ ] search through your tasks by simply using keywords in your saved tasks!

You should totally use it because 🔥:
1. It is a chatbot so you will find it very familiar to use
2. Adding, deleting and searching tasks has never been simpler!
3. Exciting new features will be constantly added so join us now!

Here's some examples of what you can tell Duke and what will be returned!:
- "todo read book" will be saved as "T | O | read book" inside Duke! "T" indicates "todo" and "O" indicates that it is not done
- "event attend wedding /at 2022-11-28 18:00" will be saved as "E | O | attend wedding | Nov 28 2022 | 18:00"
- "deadline finish homework /by 2022-05-20 07:00" will be saved as "D | O | finish homework | May 20 2022 | 07:00"

```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

joosxi committed:
Level-1

joosxi committed:
Removed unused code that was previously commented out

joosxi committed:
Level-2, added the Add and List commands functionality

joosxi committed:
removed unneeded code that was commented out

joosxi committed:
Level-3

joosxi committed:
removed unneeded code that was commented out

joosxi committed:
Level-4

joosxi committed:
Level-5, but not sure if exceptions was done correctly

joosxi committed:
Level-6

joosxi committed:
added tests for automated text UI testing, deleted commented out code from Duke.java

joosxi committed:
added code for Level-7

joosxi committed:
added code for Level-8

joosxi committed:
Merge branch 'branch-Level-7'

joosxi committed:
Merge branch 'branch-Level-8'

joosxi committed:
A-MoreOOP

joosxi committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

joosxi committed:
added all files to package 'duke'

joosxi committed:
Added JUnit tests

joosxi committed:
Added JavaDoc for all files

joosxi committed:
Edited code to follow Java coding standards

joosxi committed:
Added punctuation for JavaDocs

joosxi committed:
added code for Level-9 but should have added it in another branch

joosxi committed:
Merge branch 'branch-A-JavaDoc'

joosxi committed:
Merge branch 'branch-A-CodingStandard'

joosxi committed:
Level-10 code partially done but in master branch

joosxi committed:
Level-10 code updated

leirdas commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+    private Ui ui;
+
+    /**
+     * Constructor for the Duke class.
```
Our coding standards state that there should be an empty line between the description and parameter section.

leirdas reviewed (commented):
Overall, code is easy to read and classes are well split. I dont think there are any naming convention mistakes, variables have meaningful names which helps with understanding your code.

o-ohst commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,64 @@
+ * Represents a task that has to be done by a certain date and/or time.
+ */
+public class Deadline extends Task {
+    protected String by = "";
```
Perhaps `by` could be named as `deadline` instead?

o-ohst commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,91 @@
+        } else if (response.equals("bye")) {
+            ;
+        } else {
+            System.out.println("     Please specify one of the 3 commands before your task to add a task:\n"
```
Line is too long. Perhaps break the line.

o-ohst commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+/**
+ * A class in charge of interactions with the user.
+ */
+public class Ui {
```
I like your code, it is very clean

o-ohst commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,91 @@
+    /**
+     * Constructor for the Parser class
+     */
+    public Parser() {
```
Is this necessary?

o-ohst commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,64 @@
+ */
+public class Event extends Task {
+    protected String at = "";
+    protected LocalDate atDate;
```
Maybe can combine `LocalDate` and `LocalTime` as `LocalDateTime`.

joosxi committed:
Level-10

joosxi committed:
Merge branch 'branch-Level-10'

joosxi committed:
Did some testings for assert but reverted changes to code afterwards

joosxi committed:
Added code for C-Update

joosxi committed:
Added assert statements, but should have done it in a separate branch

joosxi committed:
Modify code to improve code quality, mainly splitting long methods

Some methods were too lengthy.

Split methods up into multiple shorter methods.

joosxi committed:
ParserTest class: Remove code in the class

Code in Parser class has been edited to be used for GUI and not
terminal.

That is why original code in ParserTest class no longer works with
current code in Parser class.

joosxi committed:
Merge pull request #1 from joosxi/branch-A-Assertions

branch-A-Assertions

joosxi committed:
Merge branch 'master' into branch-A-CodeQuality

joosxi committed:
Merge pull request #2 from joosxi/branch-A-CodeQuality

Branch-A-CodeQuality

joosxi committed:
Update README.md

joosxi committed:
Add files via upload

joosxi committed:
shifted Launcher class to java package, and added user guide and screenshot of app

joosxi committed:
merge conflict resolved

joosxi committed:
Readme file

joosxi committed:
edited readme file

joosxi committed:
Clean up dead code and edit formatting

joosxi committed:
Update code to provide more information to users when wrong format of command is given

joosxi committed:
Update README.md
